signals_windows.go

```diff
@@ -25,39 +25,136 @@ import (
 	"golang.org/x/sys/windows"
 )
 
-var signalMap = map[string]syscall.Signal{
-	"HUP":    syscall.Signal(windows.SIGHUP),
-	"INT":    syscall.Signal(windows.SIGINT),
-	"QUIT":   syscall.Signal(windows.SIGQUIT),
-	"SIGILL": syscall.Signal(windows.SIGILL),
-	"TRAP":   syscall.Signal(windows.SIGTRAP),
-	"ABRT":   syscall.Signal(windows.SIGABRT),
-	"BUS":    syscall.Signal(windows.SIGBUS),
-	"FPE":    syscall.Signal(windows.SIGFPE),
-	"KILL":   syscall.Signal(windows.SIGKILL),
-	"SEGV":   syscall.Signal(windows.SIGSEGV),
-	"PIPE":   syscall.Signal(windows.SIGPIPE),
-	"ALRM":   syscall.Signal(windows.SIGALRM),
-	"TERM":   syscall.Signal(windows.SIGTERM),
+const (
+	linuxSigrtmin = 34
+	linuxSigrtmax = 64
+)
+
+var signalMapWindows = map[string]syscall.Signal{
+	"HUP":  syscall.Signal(windows.SIGHUP),
+	"INT":  syscall.Signal(windows.SIGINT),
+	"QUIT": syscall.Signal(windows.SIGQUIT),
+	"ILL":  syscall.Signal(windows.SIGILL),
+	"TRAP": syscall.Signal(windows.SIGTRAP),
+	"ABRT": syscall.Signal(windows.SIGABRT),
+	"BUS":  syscall.Signal(windows.SIGBUS),
+	"FPE":  syscall.Signal(windows.SIGFPE),
+	"KILL": syscall.Signal(windows.SIGKILL),
+	"SEGV": syscall.Signal(windows.SIGSEGV),
+	"PIPE": syscall.Signal(windows.SIGPIPE),
+	"ALRM": syscall.Signal(windows.SIGALRM),
+	"TERM": syscall.Signal(windows.SIGTERM),
+}
+
+// manually define signals for linux since we may be running an LCOW container, but
+// the unix syscalls do not get built when running on windows
+var signalMapLinux = map[string]syscall.Signal{
```

Comment:
I'm kinda sad that this enhanced signal lookup feature won't work on linux!
(I would have to run a linux container on windows to get `RTMIN+3` to work)

Can you put this in a generic location (signals.go), should be safe because it has `linux` in the name already, so it could be used by both signals_windows and signals_linux ? Bonus for making signals_linux actually use it.